src/main/java/org/apache/sling/auth/core/impl/SlingAuthenticator.java

```diff
@@ -663,7 +663,7 @@ private AuthenticationInfo getAuthenticationInfo(HttpServletRequest request, Htt
                     if (holder.isPathRequiresHandler(path)){
                         final AuthenticationInfo authInfo = holder.extractCredentials(
                             request, response);
-
+                            log.debug("getAuthenticationInfo: Selected class {} to perform authentication", holder.getHandler().getClass().getName());
```

Comment:
Based on the EngineAuthenticationHandlerHolder there are cases where this could be null, which should be handled. How about:
```suggestion
                            log.debug("getAuthenticationInfo: Selected class {} to perform authentication", Optional.ofNullable(holder.getHandler()).map(h -> h.getClass().getName()).orElse("null"));
```